core/src/rpc.rs

```diff
@@ -1406,8 +1428,43 @@ impl RpcSol for RpcSolImpl {
         }
     }
 
-    fn send_transaction(&self, meta: Self::Metadata, data: String) -> Result<String> {
+    fn send_transaction(
+        &self,
+        meta: Self::Metadata,
+        data: String,
+        config: Option<RpcSendTransactionConfig>,
+    ) -> Result<String> {
+        let config = config.unwrap_or_default();
         let (wire_transaction, transaction) = deserialize_bs58_transaction(data)?;
+        let signature = transaction.signatures[0].to_string();
+
+        if !config.skip_preflight {
+            if transaction.verify().is_err() {
+                return Err(RpcCustomError::SendTransactionPreflightFailure {
+                    message: "Transaction signature verification failed".into(),
+                }
+                .into());
+            }
+
+            if meta.request_processor.read().unwrap().health.check() != RpcHealthStatus::Ok {
+                return Err(RpcCustomError::SendTransactionPreflightFailure {
+                    message: "RPC node is unhealthy, unable to simulate transaction".into(),
+                }
+                .into());
+            }
+
+            let bank = &*meta.request_processor.read().unwrap().bank(None)?;
+            if let Err(err) = run_transaction_simulation(&bank, &[transaction]) {
+                // Note: it's possible that the transaction simulation failed but the actual
+                // transaction would succeed. In these cases the user should use the
+                // config.skip_preflight flag
```

Comment:
It's when a client performs a second operation on an account.

eg:
1. create stake account
2. delegate stake account. <-- preflight will fail until (1) is at max confirmation since the max confirmed bank won't even see the new stake account

Once we're feeling better about optimistic confirmation, we could relax preflight to use a much newer bank than max confirmation.

Really a client doing 2 before 1 is at max confirmation is gambling right now anyway, and at risk of exposing the user to a poor experience where they get unexpected error messages back from the cluster.

Reply:
Yep, I follow. I think it would be helpful to have some of this in the comment, namely performing a 2nd operation on an account before the first has reached max commitment.

Reply:
Done